.k8s/10-zalando-postgres-ha-operator/README.md

```diff
@@ -0,0 +1,196 @@
+# Installing scalable PostgreSQL via Kubernetes operator
+
+## Prerequisites
+
+- See that there is no container named 'k3d-demo-tools'
+
+```
+docker ps -a
+```
+
+- Add two nodes (agents) to the cluster
+
+```
+k3d node create demo-agent -c demo --replicas 2
+```
+
+It's not an error: ```docker failed to remove the container 'k3d-demo-tools'```
+
+
+- Get the cluster nodes
+
+```
+k3d node list
+kubectl get nodes
+```
+
+## Steps
+
+
+### 1. Install ```postgres-operator``` via Helm Chart
+
+```
+helm repo add postgres-operator-charts https://opensource.zalando.com/postgres-operator/charts/postgres-operator
+helm repo update
+helm upgrade --install --cleanup-on-fail postgres-operator postgres-operator-charts/postgres-operator --namespace example-api --create-namespace
+```
+
+The operator and all its resources will be installed to namespace ```example-api```, which will be created if necessary
+
+
+### 2. See new Custom Resource Definitions 
+
+**From OpenLens**
+
+Custom Resources / acid.zalan.do / postgresql
+
+
+### 3. Create new PostgreSQL cluster
+
+**From CLI**
+
+```
+kubectl ns example-api
+cd ~/springboot-api-rest-example/.k8s/10-zalando-postgres-ha-operator
+kubectl apply -f db-crd.yaml
+```
+
+
+### 4. Watch how PostgreSQL cluster is creating in OpenLens 
+
+**From OpenLens**
+
+- Workloads / StatefulSets
+- Custom Resources / acid.zalan.do / postgresql
+
+
+### 5. How PostgreSQL cluster manifest was created? 
+
+**Install ```postgres-operator-ui``` via Helm Chart**
+
+```
+helm repo add postgres-operator-ui-charts https://opensource.zalando.com/postgres-operator/charts/postgres-operator-ui
+helm repo update
+helm upgrade --install --cleanup-on-fail postgres-operator-ui postgres-operator-ui-charts/postgres-operator-ui -f postgres-operator-ui_values.yaml --namespace example-api --create-namespace
+```
+
+**Remark**. 
+Value of ```envs.targetNamespace``` changed to ```'*'``` in the file ```postgres-operator-ui_values.yaml```
+
+
+**Open ```postgres-operator-ui``` in browser**
+
+Network / Services --> namespace: 'example-api', service: 'postgres-operator-ui'
+
+Forward port, open in browser the web console
+
+**From browser on local machine**
+
+Go to Tab 'New cluster', fillout the fields
+
+	Name: pg-demo
+	Namespace: choose 'example-api' (available because of targetNamespace: '*' for the ui)
+	Owning team: leave the default 'acid', it will be a prefix for Kubernetes resources names
+	Number of instances: 3
+	Master load balancer: uncheck 
+	Replica load balancer: uncheck
+	Enable connection pool: check
+	Volume size: 1 (GiB)
+	+ Users: testuser
+	+ Databases: testdb
+	Resources: 100,500,100,1000
+
+Click 'Validate' in the up-right
+
+See the manifest in the left pane 'Cluster YAML definition'
```

Comment:
I believe the whole step no. 5 intersects with steps 1-4. There may be misunderstanding. We need to make it clear that there are completely analogous separate ways to create a Postgres cluster, and that we do not need to follow the steps for both of these ways one after another. Either, or. Now it is not clear enough IMHO. And what is the difference between postgres-operator and postgres-operator-ui? We can have both of them installed?

Reply:
We apply manifest for PostgreSQL cluster on Step 3.
Step 5 explains how this manifest was created. We were using another tool, postgres-operator-ui, for creating the manifest.
If you think that there may be misunderstanding, I delete the step 5.